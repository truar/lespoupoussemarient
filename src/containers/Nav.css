nav.bg-dark {
    background-color: rgba(0,0,0,0.7)!important;
}

nav.desktop {
    display: none;
}

@media (min-width: 992px) {
    body nav.mobile {
        display: none;
    }
    body nav.desktop.navbar {
        display: flex;
        position:fixed;
        left: -260px;
        top: 0;
        width:350px;
        background-color:rgba(0,0,0,0.5);
        height:100%;
        z-index:100;
        transition-property: left, background-color;
        transition-delay: 0s;
        transition-duration: 0.4s;
        transition-timing-function: ease-in;
        font-size: 2rem;
        text-align: right;
        font-family: SignPainter;
        padding: 0;
    }
    
    body nav.desktop.navbar:hover {
        transition-property: left, background-color;
        transition-duration: 0.4s;
        transition-timing-function: ease-in;
        transition-delay: 0s;
        left: 0px;
        background-color:rgba(0,0,0,0.7);
    }

    nav.desktop>button, nav.desktop>a.navbar-brand {
        display: none;
    }

    nav.desktop, nav.desktop>ul {
        display: flex!important;
        height:100%;
        width: 100%;
    }

    nav.desktop ul {
        justify-content: space-evenly;
    }

    nav.desktop ul li a {
        color: white!important;
    }

    nav.desktop ul li a:hover {
        color: rgb(89,89,89)!important;
    }

    #user_img {
        background-image: url('../images/icons/user.png');
        background-position: center top;
    }
    
    .selected #user_img, nav ul li a:hover #user_img {
        background-image: url('../images/icons/user_select.png');
    }

    nav.desktop span.img_nav {
        width: 40px;
        height: 55px;
        vertical-align: middle;
        margin-left: 30px;
        margin-right: 30px;
        display: inline-block;
        background-size: 100%;
        background-repeat: no-repeat;
        background-position: center center;
    }

    
}