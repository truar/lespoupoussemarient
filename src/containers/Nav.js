import React, {Component} from 'react';
import { Collapse, Navbar, NavbarToggler, NavbarBrand, Nav as NavStrap, NavItem, NavLink } from 'reactstrap';
import './Nav.css';

class Nav extends Component {
  constructor(props) {
    super(props);

    this.toggleNavbar = this.toggleNavbar.bind(this);
    this.navItems = props.navItems;
    this.state = {
      collapsed: true,
    };
  }

  toggleNavbar() {
    this.setState({
      collapsed: !this.state.collapsed
    });
  }

  render() {
    const renderNavItems = this.navItems.map((item, i) => {
      return (
        <NavItem key={i}>
          <NavLink onClick={this.toggleNavbar} href={item.href}>{item.label} <span id={item.spanId} className='img_nav' /></NavLink>
        </NavItem>
      );
    });

    const {selectedNavItem} = this.props;

    return (
      <>
        <Navbar color="dark" className="mobile" dark fixed="top">
          <NavbarBrand href="/" className="mr-auto">{selectedNavItem.label}</NavbarBrand>
          <NavbarToggler onClick={this.toggleNavbar} className="mr-2" />
          <Collapse isOpen={!this.state.collapsed} navbar>
            <NavStrap navbar>
                {renderNavItems}
            </NavStrap>
          </Collapse>
        </Navbar>
        <Navbar color="dark" className="desktop" dark fixed="top">
          <NavbarBrand href="/" className="mr-auto">{selectedNavItem.label}</NavbarBrand>
          <NavbarToggler onClick={this.toggleNavbar} className="mr-2" />
          <NavStrap navbar>
              {renderNavItems}
          </NavStrap>
        </Navbar>
      </>
    );
  }
}

export default Nav;