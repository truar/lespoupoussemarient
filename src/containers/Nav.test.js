import React from 'react';
import Nav from './Nav';
import { shallow } from 'enzyme';

const navItems = [{
  href: 'test', 
  label: 'test'
}];
it('renders without crashing', () => {
  const component = shallow(<Nav selectedNavItem={navItems[0]} navItems={navItems} />);
  expect(component).toMatchSnapshot();
});
