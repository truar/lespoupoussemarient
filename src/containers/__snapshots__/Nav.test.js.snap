// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav
    navItems={
      Array [
        Object {
          "href": "test",
          "label": "test",
        },
      ]
    }
    selectedNavItem={
      Object {
        "href": "test",
        "label": "test",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navbar
          className="mobile"
          color="dark"
          dark={true}
          expand={false}
          fixed="top"
          tag="nav"
        >
          <NavbarBrand
            className="mr-auto"
            href="/"
            tag="a"
          >
            test
          </NavbarBrand>
          <NavbarToggler
            className="mr-2"
            onClick={[Function]}
            tag="button"
            type="button"
          />
          <Collapse
            appear={false}
            enter={true}
            exit={true}
            in={false}
            isOpen={false}
            mountOnEnter={false}
            navbar={true}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            tag="div"
            timeout={350}
            unmountOnExit={false}
          >
            <Nav
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  href="test"
                  onClick={[Function]}
                  tag="a"
                >
                  test
                </NavLink>
              </NavItem>
            </Nav>
          </Collapse>
        </Navbar>,
        <Navbar
          className="desktop"
          color="dark"
          dark={true}
          expand={false}
          fixed="top"
          tag="nav"
        >
          <Nav
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                active={true}
                href="test"
                tag="a"
              >
                test
                 
                <span
                  className="img_nav"
                />
              </NavLink>
            </NavItem>
          </Nav>
        </Navbar>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <NavbarBrand
              className="mr-auto"
              href="/"
              tag="a"
            >
              test
            </NavbarBrand>,
            <NavbarToggler
              className="mr-2"
              onClick={[Function]}
              tag="button"
              type="button"
            />,
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="test"
                    onClick={[Function]}
                    tag="a"
                  >
                    test
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>,
          ],
          "className": "mobile",
          "color": "dark",
          "dark": true,
          "expand": false,
          "fixed": "top",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "test",
              "className": "mr-auto",
              "href": "/",
              "tag": "a",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "mr-2",
              "onClick": [Function],
              "tag": "button",
              "type": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "appear": false,
              "children": <Nav
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="test"
                    onClick={[Function]}
                    tag="a"
                  >
                    test
                  </NavLink>
                </NavItem>
              </Nav>,
              "enter": true,
              "exit": true,
              "in": false,
              "isOpen": false,
              "mountOnEnter": false,
              "navbar": true,
              "onEnter": [Function],
              "onEntered": [Function],
              "onEntering": [Function],
              "onExit": [Function],
              "onExited": [Function],
              "onExiting": [Function],
              "tag": "div",
              "timeout": 350,
              "unmountOnExit": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="test"
                      onClick={[Function]}
                      tag="a"
                    >
                      test
                    </NavLink>
                  </NavItem>,
                ],
                "navbar": true,
                "tag": "ul",
                "vertical": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "0",
                  "nodeType": "function",
                  "props": Object {
                    "children": <NavLink
                      href="test"
                      onClick={[Function]}
                      tag="a"
                    >
                      test
                    </NavLink>,
                    "tag": "li",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "test",
                      "href": "test",
                      "onClick": [Function],
                      "tag": "a",
                    },
                    "ref": null,
                    "rendered": "test",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Nav
            navbar={true}
            tag="ul"
            vertical={false}
          >
            <NavItem
              tag="li"
            >
              <NavLink
                active={true}
                href="test"
                tag="a"
              >
                test
                 
                <span
                  className="img_nav"
                />
              </NavLink>
            </NavItem>
          </Nav>,
          "className": "desktop",
          "color": "dark",
          "dark": true,
          "expand": false,
          "fixed": "top",
          "tag": "nav",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavItem
                tag="li"
              >
                <NavLink
                  active={true}
                  href="test"
                  tag="a"
                >
                  test
                   
                  <span
                    className="img_nav"
                  />
                </NavLink>
              </NavItem>,
            ],
            "navbar": true,
            "tag": "ul",
            "vertical": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "children": <NavLink
                  active={true}
                  href="test"
                  tag="a"
                >
                  test
                   
                  <span
                    className="img_nav"
                  />
                </NavLink>,
                "tag": "li",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": true,
                  "children": Array [
                    "test",
                    " ",
                    <span
                      className="img_nav"
                    />,
                  ],
                  "href": "test",
                  "tag": "a",
                },
                "ref": null,
                "rendered": Array [
                  "test",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "img_nav",
                      "id": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "span",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navbar
            className="mobile"
            color="dark"
            dark={true}
            expand={false}
            fixed="top"
            tag="nav"
          >
            <NavbarBrand
              className="mr-auto"
              href="/"
              tag="a"
            >
              test
            </NavbarBrand>
            <NavbarToggler
              className="mr-2"
              onClick={[Function]}
              tag="button"
              type="button"
            />
            <Collapse
              appear={false}
              enter={true}
              exit={true}
              in={false}
              isOpen={false}
              mountOnEnter={false}
              navbar={true}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              tag="div"
              timeout={350}
              unmountOnExit={false}
            >
              <Nav
                navbar={true}
                tag="ul"
                vertical={false}
              >
                <NavItem
                  tag="li"
                >
                  <NavLink
                    href="test"
                    onClick={[Function]}
                    tag="a"
                  >
                    test
                  </NavLink>
                </NavItem>
              </Nav>
            </Collapse>
          </Navbar>,
          <Navbar
            className="desktop"
            color="dark"
            dark={true}
            expand={false}
            fixed="top"
            tag="nav"
          >
            <Nav
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  active={true}
                  href="test"
                  tag="a"
                >
                  test
                   
                  <span
                    className="img_nav"
                  />
                </NavLink>
              </NavItem>
            </Nav>
          </Navbar>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <NavbarBrand
                className="mr-auto"
                href="/"
                tag="a"
              >
                test
              </NavbarBrand>,
              <NavbarToggler
                className="mr-2"
                onClick={[Function]}
                tag="button"
                type="button"
              />,
              <Collapse
                appear={false}
                enter={true}
                exit={true}
                in={false}
                isOpen={false}
                mountOnEnter={false}
                navbar={true}
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                tag="div"
                timeout={350}
                unmountOnExit={false}
              >
                <Nav
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="test"
                      onClick={[Function]}
                      tag="a"
                    >
                      test
                    </NavLink>
                  </NavItem>
                </Nav>
              </Collapse>,
            ],
            "className": "mobile",
            "color": "dark",
            "dark": true,
            "expand": false,
            "fixed": "top",
            "tag": "nav",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "test",
                "className": "mr-auto",
                "href": "/",
                "tag": "a",
              },
              "ref": null,
              "rendered": "test",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "mr-2",
                "onClick": [Function],
                "tag": "button",
                "type": "button",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "appear": false,
                "children": <Nav
                  navbar={true}
                  tag="ul"
                  vertical={false}
                >
                  <NavItem
                    tag="li"
                  >
                    <NavLink
                      href="test"
                      onClick={[Function]}
                      tag="a"
                    >
                      test
                    </NavLink>
                  </NavItem>
                </Nav>,
                "enter": true,
                "exit": true,
                "in": false,
                "isOpen": false,
                "mountOnEnter": false,
                "navbar": true,
                "onEnter": [Function],
                "onEntered": [Function],
                "onEntering": [Function],
                "onExit": [Function],
                "onExited": [Function],
                "onExiting": [Function],
                "tag": "div",
                "timeout": 350,
                "unmountOnExit": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <NavItem
                      tag="li"
                    >
                      <NavLink
                        href="test"
                        onClick={[Function]}
                        tag="a"
                      >
                        test
                      </NavLink>
                    </NavItem>,
                  ],
                  "navbar": true,
                  "tag": "ul",
                  "vertical": false,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": "0",
                    "nodeType": "function",
                    "props": Object {
                      "children": <NavLink
                        href="test"
                        onClick={[Function]}
                        tag="a"
                      >
                        test
                      </NavLink>,
                      "tag": "li",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "test",
                        "href": "test",
                        "onClick": [Function],
                        "tag": "a",
                      },
                      "ref": null,
                      "rendered": "test",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Nav
              navbar={true}
              tag="ul"
              vertical={false}
            >
              <NavItem
                tag="li"
              >
                <NavLink
                  active={true}
                  href="test"
                  tag="a"
                >
                  test
                   
                  <span
                    className="img_nav"
                  />
                </NavLink>
              </NavItem>
            </Nav>,
            "className": "desktop",
            "color": "dark",
            "dark": true,
            "expand": false,
            "fixed": "top",
            "tag": "nav",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <NavItem
                  tag="li"
                >
                  <NavLink
                    active={true}
                    href="test"
                    tag="a"
                  >
                    test
                     
                    <span
                      className="img_nav"
                    />
                  </NavLink>
                </NavItem>,
              ],
              "navbar": true,
              "tag": "ul",
              "vertical": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "function",
                "props": Object {
                  "children": <NavLink
                    active={true}
                    href="test"
                    tag="a"
                  >
                    test
                     
                    <span
                      className="img_nav"
                    />
                  </NavLink>,
                  "tag": "li",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": true,
                    "children": Array [
                      "test",
                      " ",
                      <span
                        className="img_nav"
                      />,
                    ],
                    "href": "test",
                    "tag": "a",
                  },
                  "ref": null,
                  "rendered": Array [
                    "test",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "img_nav",
                        "id": undefined,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "span",
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
